pkg/safenetlink/netlink_linux.go

```diff
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package safenetlink
+
+import (
+	"context"
+	"errors"
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netlink/nl"
+
+	"github.com/cilium/cilium/pkg/resiliency"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+const (
+	netlinkRetryInterval = 1 * time.Millisecond
+	netlinkRetryMax      = 30
+)
+
+// withRetry runs the netlinkFunc. If netlinkFunc returns netlink.ErrDumpInterrupted, the function is retried up
+// to 20 times. If success or any other error is returned, withRetry returns immediately, propagating the error.
```

Comment:
nit: The constant is already wrong, drop it to be consistent
```suggestion
// withRetry runs the netlinkFunc. If netlinkFunc returns netlink.ErrDumpInterrupted, the function is retried. If success or any other error is returned, withRetry returns immediately, propagating the error.
```